packages/react-scripts/package.json

```diff
@@ -62,6 +63,7 @@
     "sass-loader": "7.1.0",
     "style-loader": "0.23.0",
     "terser-webpack-plugin": "1.1.0",
+    "typescript": "^3.1.1",
```

Comment:
Remove this dependency and add a helpful warning when it's not installed, e.g.

> TypeScript has been detected in your project but no errors will be reported.
> To start reporting errors, please install the typescript package by running ....

We need to send a PR to `fork-ts-checker-webpack-plugin` dropping the peerDependency requirement and maybe adding the above error message directly to their project (for tslint and typescript).

e.g. https://github.com/Realytics/fork-ts-checker-webpack-plugin/issues/102

Reply:
@Timer when I don't install typescript inside react-scripts `fork-ts-checker-webpack-plugin` doesn't work, even if I install typescript in my project. So need to fix that in their side before removing from cra.

```shell
$ react-scripts start
module.js:549
    throw err;
    ^

Error: Cannot find module 'typescript'
    at Function.Module._resolveFilename (module.js:547:15)
    at Function.Module._load (module.js:474:25)
    at Module.require (module.js:596:17)
    at require (internal/module.js:11:18)
    at Object.<anonymous> (/Users/brunolemos/Projects/packages/create-react-app/node_modules/fork-ts-checker-webpack-plugin/lib/CancellationToken.js:6:10)
    at Module._compile (module.js:652:30)
    at Object.Module._extensions..js (module.js:663:10)
    at Module.load (module.js:565:32)
    at tryModuleLoad (module.js:505:12)
    at Function.Module._load (module.js:497:3)
error Command failed with exit code 1.
```

Reply:
Yeah, let's fix that on their end. We can't ship `typescript` by default -- it's nearly 50MB for a feature someone might not use.